app/scripts/lib/ppom/ppom-middleware.ts

```diff
@@ -58,15 +63,45 @@ export function createPPOMMiddleware(
       const { chainId } = networkController.state.providerConfig;
       if (
         securityAlertsEnabled &&
-        ConfirmationMethods.includes(req.method) &&
+        CONFIRMATION_METHODS.includes(req.method) &&
         SUPPORTED_CHAIN_IDS.includes(chainId)
       ) {
         // eslint-disable-next-line require-atomic-updates
-        req.securityAlertResponse = await ppomController.usePPOM(
-          async (ppom: PPOM) => {
-            return ppom.validateJsonRpc(req);
-          },
-        );
+        const securityAlertId = uuid();
+
+        ppomController.usePPOM(async (ppom: PPOM) => {
+          try {
+            const securityAlertResponse = await ppom.validateJsonRpc(req);
+            securityAlertResponse.securityAlertId = securityAlertId;
+            updateSecurityAlertResponseByTxId(req, securityAlertResponse);
+          } catch (error: any) {
+            sentry?.captureException(error);
+            console.error('Error validating JSON RPC using PPOM: ', error);
+            req.securityAlertResponse = {
+              result_type: BlockaidResultType.Failed,
+              reason: BlockaidReason.failed,
+              description:
+                'Validating the confirmation failed by throwing error.',
+            };
+          }
+        });
+
+        if (SIGNING_METHODS.includes(req.method)) {
+          req.securityAlertResponse = {
+            securityAlertId,
+          };
+          appStateController.addSignatureSecurityAlertResponse({
+            reason: 'loading',
+            result_type: 'validation_in_progress',
+            securityAlertId,
+          });
+        } else {
+          req.securityAlertResponse = {
+            reason: 'loading',
+            result_type: 'validation_in_progress',
+            securityAlertId,
+          };
+        }
       }
     } catch (error: any) {
       sentry?.captureException(error);
```

Comment:
Not important in this PR, but later we can refactor out validation from here as middleware is not needed for async validation - this can be done in controllers also.